AngularJS/src/components/registration/registration.component.js

```diff
@@ -2,18 +2,28 @@ const angular = require('angular');
 
 angular.module('app-bootstrap').component('registration', {
   template: require('./registration.pug')(),
-  controller: ['userService', '$state', function (userService, $state) {
-    this.submitted = false;
+  controller: ['userService', 'sessionService', '$state', function (userService, sessionService, $state) {
     this.user = {
-      name: '',
-      lastname: '',
+      first_name: '',
+      last_name: '',
       password: '',
-      email: ''
+      password_confirmation: '',
+      email: '',
+      locale: 'es'
     };
-    this.createUser = () => {
-      this.submitted = true;
-      userService.create(this.user);
-      $state.transitionTo('navbar.home');
+    this.createUser = (validForm) => {
+      if (validForm) {
+        userService.create(this.user)
+        .then((response) => {
+          sessionService.setUserInfo(response.data);
+          $state.transitionTo('navbar.home');
+        })
+        .catch((erro) => {
+          if (erro.status === 401) {
```

Comment:
401 is `Unauthorized` so being the sign up I don't think this specific error will happen. You can delete this if and set to `true` the `loginAuthError`